stdlib/messaging/rabbitmq/src/main/ballerina/src/rabbitmq/Module.md

```diff
@@ -1,99 +1,169 @@
+```ballerina
+   var exchangeResult = newChannel->exchangeDeclare({ exchangeName: "MyExchange",
+                                                      exchangeType: rabbitmq:DIRECT_EXCHANGE,
+                                                      durable: true,
+                                                      autoDelete: true });
+   if (exchangeResult is error) {
+        io:println("An error occurred while declaring the exchange");
+   }
+   
+   var queueResult = newChannel->queueDeclare({ queueName: "MyQueue", 
+                                                durable: true,
+                                                exclusive: false,
+                                                autoDelete: false });
+   if (queueResult is error) {
+    io:println("An error occurred while creating the MyQueue queue.");
+   }
+
+    var bindResult = newChannel.queueBind("MyQueue", "MyExchange", "routing-key");
+        if (bindResult is error) {
+            io:println("Error ocurred while binding the queue to the exhchange");
         }
-    }
-    _ = wait {w1, w2, w3};
-}
 ```
-### RabbitMQ Subscriber
 
-Following program will consume a message from a RabbitMQ server
+This sample code will declare,
+ - A durable, auto-delete exchange of type rabbitmq:DIRECT_EXCHANGE
+ - A durable, non-exclusive, non-auto-delete queue with a well-known name
+
+#### Deleting entities and purging queues
+
+- Delete a queue
+```ballerina
+   newChannel->queueDelete("MyQueue");
+```
+- Delete a queue only if it is empty
+```ballerina
+   newChannel->queueDelete("MyQueue", false, true);
+```
+- Delete a queue only if it is unused (does not have any consumers)
+```ballerina
+   newChannel->queueDelete("MyQueue", true, false);
+```
+- Delete an exchange
+```ballerina
+   newChannel->exchangeDelete("MyExchange");
+```
+- A queue can be purged (all of its messages deleted):
+```ballerina
+   newChannel->queuePurge("MyQueue");
+```
+
+### Publishing Messages
+
+To publish a message to an exchange, use the basicPublish() function as follows:
 
 ```ballerina
-import ballerina/log;
-import ballerina/rabbitmq;
+   var sendResult = newChannel->basicPublish("Hello from Ballerina", "MyQueue");
+   if (sendResult is error) {
+        io:println("An error occurred while sending the message to MyQueue using newChannel.");
+   }
+``` 
+Setting other properties of the message such as routing headers can be done by using the `BasicProperties` record with appropriate values. 
+
+### Consuming Messages using Consumer Services
 
-// Creates a ballerina RabbitMQ connection that allows re-usability if necessary.
-rabbitmq:Connection connection = new({ host: "localhost", port: 5672 });
+The most efficient way to receive messages is to set up a subscription using a Ballerina RabbitMQ `Listener` and any number of consumer services. The messages will then be delivered automatically as they arrive, rather than having to be explicitly requested. 
```

Comment:
```suggestion
The most efficient way to receive messages is to set up a subscription using a Ballerina RabbitMQ `Listener` and any number of consumer services. The messages will then be delivered automatically as they arrive rather than having to be explicitly requested. 
```